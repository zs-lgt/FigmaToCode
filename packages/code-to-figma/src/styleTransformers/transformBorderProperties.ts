import { BorderProps, CornerProps, GeometryProps } from '../types';
import { transformSize, TSize } from '../helpers/size';
import { colorToPaint } from './transformColors';

export interface BorderStyleProperties {
    borderColor: string;
    borderWidth: number;

    borderRadius: TSize;
    borderBottomLeftRadius: TSize;
    borderBottomRightRadius: TSize;
    borderTopLeftRadius: TSize;
    borderTopRightRadius: TSize;
}

export const transformBorderStyleProperties = (
    styles?: Partial<BorderStyleProperties>
): BorderProps & GeometryProps => {
    if (!styles) {
        return {};
    }

    const props: BorderProps & GeometryProps & CornerProps = {};

    if (styles.borderRadius) {
        props.cornerRadius = transformSize(styles.borderRadius);
    }

    if (styles.borderTopLeftRadius) {
        props.topLeftRadius = transformSize(styles.borderTopLeftRadius);
    }

    if (styles.borderTopRightRadius) {
        props.topRightRadius = transformSize(styles.borderTopRightRadius);
    }

    if (styles.borderBottomLeftRadius) {
        props.bottomLeftRadius = transformSize(styles.borderBottomLeftRadius);
    }

    if (styles.borderBottomRightRadius) {
        props.bottomRightRadius = transformSize(styles.borderBottomRightRadius);
    }

    if (styles.borderColor) {
        props.strokes = [colorToPaint(styles.borderColor)];
    }

    if (styles.borderWidth) {
        props.strokeWeight = transformSize(styles.borderWidth);
    }

    return props;
};
