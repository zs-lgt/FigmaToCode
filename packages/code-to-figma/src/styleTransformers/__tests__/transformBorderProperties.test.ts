import { transformBorderStyleProperties } from '../transformBorderProperties';

describe('transformBorderProperties', () => {
    it('borderRadius', () => {
        const result = transformBorderStyleProperties({ borderRadius: 5 });
        expect(result).toMatchSnapshot();
    });

    it('borderTopLeftRadius', () => {
        const result = transformBorderStyleProperties({ borderTopLeftRadius: 5 });
        expect(result).toMatchSnapshot();
    });

    it('borderTopRightRadius', () => {
        const result = transformBorderStyleProperties({ borderTopRightRadius: 5 });
        expect(result).toMatchSnapshot();
    });

    it('borderBottomLeftRadius', () => {
        const result = transformBorderStyleProperties({ borderBottomLeftRadius: 5 });
        expect(result).toMatchSnapshot();
    });

    it('borderBottomRightRadius', () => {
        const result = transformBorderStyleProperties({ borderBottomRightRadius: 5 });
        expect(result).toMatchSnapshot();
    });
    it('borderColor', () => {
        const result = transformBorderStyleProperties({ borderColor: '#ffffff' });
        expect(result).toMatchSnapshot();
    });
    it('borderWidth', () => {
        const result = transformBorderStyleProperties({ borderWidth: 2 });
        expect(result).toMatchSnapshot();
    });
});
