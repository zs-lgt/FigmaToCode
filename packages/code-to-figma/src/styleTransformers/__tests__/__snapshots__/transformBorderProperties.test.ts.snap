// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformBorderProperties borderBottomLeftRadius 1`] = `
Object {
  "bottomLeftRadius": 5,
}
`;

exports[`transformBorderProperties borderBottomRightRadius 1`] = `
Object {
  "bottomRightRadius": 5,
}
`;

exports[`transformBorderProperties borderColor 1`] = `
Object {
  "strokes": Array [
    Object {
      "color": Object {
        "b": 1,
        "g": 1,
        "r": 1,
      },
      "opacity": 1,
      "type": "SOLID",
    },
  ],
}
`;

exports[`transformBorderProperties borderRadius 1`] = `
Object {
  "cornerRadius": 5,
}
`;

exports[`transformBorderProperties borderTopLeftRadius 1`] = `
Object {
  "topLeftRadius": 5,
}
`;

exports[`transformBorderProperties borderTopRightRadius 1`] = `
Object {
  "topRightRadius": 5,
}
`;

exports[`transformBorderProperties borderWidth 1`] = `
Object {
  "strokeWeight": 2,
}
`;
